CMakeLists.txt

```diff
@@ -72,6 +72,14 @@ target_compile_definitions(mbed-core
         ${MBED_CONFIG_DEFINITIONS}
 )
 
+# Backward compatibility with mbed-cli 1
+if(BUILD_TESTING)
```

Comment:
`include(CTest)` will automatically set `BUILD_TESTING` to `ON`. If application `CMakeLists.txt` wrongly includes ctest will automatically set BUILD_TESTING

Reply:
a user can always overwrite the build testing macro. I'll wait until unittests gets in (it will use project name), I'll mark this as needs preceding PR